File: payola/s2js/runtime/src/main/scala/s2js/runtime/scala/collection/MapCompanion.scala
package s2js.runtime.scala.collection

import s2js.compiler.NativeJs

trait MapCompanion
{
    def empty: Map[Any, Any]

    def apply(xs: Any*): Map[Any, Any] = {
        var m = empty
        xs.foreach(m += _)
        m
    }
}

File: payola/web/server/public/javascripts/runtime/s2js/runtime/scala/collection/MapCompanion.js
goog.provide('scala.collection.MapCompanion');
goog.require('scala.collection.immutable.List');
scala.collection.MapCompanion = function() {
var self = this;
};
scala.collection.MapCompanion.prototype.$apply = function() {
var self = this;
var xs = scala.collection.immutable.List.fromJsArray([].splice.call(arguments, 0, arguments.length - 0));
var m = self.empty();
xs.foreach(function($x$1) {
m.$plus$eq($x$1);
});
return m;

};
scala.collection.MapCompanion.prototype.__class__ = new s2js.Class('scala.collection.MapCompanion', []);
